tools/ConvertTo-PseudoLocalization.ps1

```diff
@@ -98,22 +72,86 @@ foreach ($entry in $content.root.data) {
     foreach ($char in $value.ToCharArray()) {
         if ($m = $mapping[$char]) {
             $newValue += $m[$rng.Next(0, $mapping[$char].Length)]
-        } else {
+        }
+        else {
             $newValue += $char
         }
     }
 
-    # Replace all placeholders with their original values
-    foreach ($kv in $placeholders.GetEnumerator()) {
-        $newValue = $newValue.Replace($kv.Key, $kv.Value)
-    }
+    return $newValue
+}
 
+function PadPseudoLocalization([string]$value) {
     # Add 40% padding to the end of the string
-    $paddingLength = [System.Math]::Round(0.4 * $newValue.Length)
+    $paddingLength = [System.Math]::Round(0.4 * $value.Length)
     $padding = ' !!!' * ($paddingLength / 4 + 1)
-    $newValue += $padding.Substring(0, $paddingLength)
+    $value += $padding.Substring(0, $paddingLength)
+    return $value
+}
+
+if ($path.EndsWith(".resw")) {
+    foreach ($entry in $content.SelectNodes('//root/data')) {
+        $value = $entry.value
+        $comment = $entry.SelectSingleNode('comment')?.'#text' ?? ''
+        $placeholders = @{}
+
+        if ($comment.StartsWith('{Locked')) {
+            # Skip {Locked} and {Locked=qps-ploc} entries
+            if ($comment -match '\{Locked(\}|=[^}]*qps-ploc).*') {
+                continue
+            }
+
+            $lockedList = ($comment -replace '\{Locked=(.*?)\}.*', '$1') -split ','
+            $placeholderChar = 0xE000
 
-    $entry.value = $newValue
+            # Replaced all locked words with placeholders from the Unicode Private Use Area
+            foreach ($locked in $lockedList) {
+                if ($locked.StartsWith('"') -and $locked.EndsWith('"')) {
+                    $locked = $locked.Substring(1, $locked.Length - 2)
+                    $locked = $locked.Replace('\"', '"')
+                }
+
+                $placeholder = "$([char]$placeholderChar)"
+                $placeholderChar++
+
+                $placeholders[$placeholder] = $locked
+                $value = $value.Replace($locked, $placeholder)
+            }
+        }
+
+        $newValue = GetPseudoLocalization $entry.name $value
+
+        # Replace all placeholders with their original values
+        foreach ($kv in $placeholders.GetEnumerator()) {
+            $newValue = $newValue.Replace($kv.Key, $kv.Value)
+        }
+
+        $newValue = PadPseudoLocalization $newValue
+
+        $entry.value = $newValue
+    }
+} elseif ($path.EndsWith(".xml")) {
+    foreach ($entry in $content.DocumentElement.SelectNodes('//*[@_locID]/text()')) {

```

Comment:
technically `Locked` exists in this format too; do we not have any?

Reply:
Ah, we actually do.

```57:    <!-- _locComment_text="{MaxLength=1500} {Locked=__VERSION_NUMBER__} App Release Note" -->Version __VERSION_NUMBER```

Reply:
Ah f... Let me try to fix that.